File: app/src/main/java/ca/ivasenko/mobiletest/categories/ui/CategoryListViewModel.java
package ca.ivasenko.mobiletest.categories.ui;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import java.util.List;

import ca.ivasenko.mobiletest.categories.data.CategoryRepository;
import ca.ivasenko.mobiletest.categories.model.Category;

public class CategoryListViewModel extends AndroidViewModel
{
    private final CategoryRepository mRepository;
    private LiveData<List<Category>> mAllCategories;

    public CategoryListViewModel(@NonNull Application application)
    {
        super(application);
        mRepository = new CategoryRepository(application);
        mAllCategories = mRepository.getAllCategories();
    }

    LiveData<List<Category>> getAllCategories()
    {
        return mAllCategories;
    }
}

File: app/src/main/java/com/rodionov/igor/mobiletest/viewModel/CategoryListViewModel.kt
package com.rodionov.igor.mobiletest.viewModel

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel

import com.rodionov.igor.mobiletest.model.Category
import com.rodionov.igor.mobiletest.repository.CategoriesRepository

class CategoryListViewModel(repository: CategoriesRepository) : ViewModel() {

    val categoryList: LiveData<List<Category>>

    init {
        categoryList = repository.categoryListData
    }
}
